### DISCUSSION TITLE
Porting WAMR to VEX V5

### DISCUSSION THREAD

**Discussion by lewisfm** [bytecodealliance/wasm-micro-runtime#3952] [Q&A]
Hey there! I'd like to use WAMR on the VEX V5, a single-core, bare-metal environment built on a Cortex A9. C programs are compiled with `arm-none-eabi-gcc`, and notably, the platform has no concept of threads or processes. How feasible would it be for me to port WAMR's AOT mode to this platform and embed it in existing Rust code, given that it depends on the package "Threads" in CMakeLists.txt?

**Comment by lum1n0us** [comment]
The threads library isn't always necessary on all platforms. You might find something useful in the *CMakeLists.txt* files located in the *product-mini/platforms* directory. The top-level *CMakeLists.txt* is a general example and may not be suitable for your specific case.
